connector/datadogconnector/connector.go

```diff
@@ -42,6 +42,9 @@ type traceToMetricConnector struct {
 
 	// exit specifies the exit channel, which will be closed upon shutdown.
 	exit chan struct{}
+
+	// isStarted tracks whether Start() has been called.
+	isStarted bool
```

Comment:
Should this be an atomic? I am not sure what are the expectations regarding thread safety :thinking:

Reply:
This gets written to in `Start` and Read in `Shutdown`, so I guess the question is whether these two functions can be called concurrently or if Start can get called concurrently.

 The docs for Shutdown [mention](https://github.com/open-telemetry/opentelemetry-collector/blob/7f138124ec05d5a2096eb41d20fd4e77e8d5cf61/component/component.go#L51-L53):
```
	// This method must be safe to call:
	//   - without Start() having been called
	//   - if the component is in a shutdown state already
        ...
	Shutdown(ctx context.Context) error
```
It doesn't mention that `Shutdown` should be safe to call concurrently with `Start`.

For `Start`, the documentation doesn't mention anything about concurrency either.

So I think this should be fine ? It may be good to get this clarified in the Component docs.

Reply:
Agreed, we can handle that separately. Merging...